workmail-message-flow-state-machine/README.md

```diff
@@ -4,43 +4,34 @@ This application enables you to use AWS Step Functions to orchestrate multiple A
 
 ## Setup
 1. Deploy this application via [AWS Serverless Application Repository](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:489970191081:applications~workmail-message-flow-state-machine).
+    1. [Optional] Enter `MachineStateForOutput` to specify state of the Step Function whose output should be returned back to WorkMail. By default, output of the succeeded execution event is used.
+    2. [Optional] Enter `WaitTimeForExecution` to specify number of seconds to wait for result after the state machine is first invoked, useful when step function finishes within few seconds. By default, value is 1 second.
 2. Open the [WorkMail Console](https://console.aws.amazon.com/workmail/) and create a synchronous **RunLambda** [Email Flow Rule](https://docs.aws.amazon.com/workmail/latest/adminguide/lambda.html#synchronous-rules) that uses this Lambda function.
+    1. `Rule timeout` must be more than the maximum duration of the step function execution. See [this section](#How timeouts and retry logic are part of this solution) for more details.
 3. Open the [AWS Step Functions Console](https://console.aws.amazon.com/states/) and modify the new state machine with your business logic. 
 
 To further customize your Lambda function, open the [AWS Lambda Console](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions) to edit and test your Lambda function with the built-in code editor.
 
-[Optional] Modify the MACHINE_STATE_FOR_OUTPUT environment variable to control which state from the Step Function execution should be returned back to the WorkMail Message Flow API. This allows you to build more asynchronous capabilities into your state machine without delaying message delivery.
+`MachineStateForOutput` allows you to build more asynchronous capabilities into your state machine without delaying message delivery.
 
-[Optional] Modify the WAIT_TIME_FOR_EXECUTION environment variable to be the number of seconds to wait after the state machine execution to look for a result. If the result is not found then it will be found in subsequent retries of the function. Use this only if your state machine is known to always execute quickly and the benefit of waiting is worth more than the extra cost of having the first function iteration running for a longer period of time.
+Use `WaitTimeForExecution` only if your state machine is known to always execute within few seconds so that the benefit of waiting is worth more than the delay caused due to retury logic.
```

Comment:
...always execute within *a* few seconds...

... the delay caused due to **retry** logic.